### PR TITLE
Detect ProvisionedThroughputExceededException and retry

### PR DESCRIPTION
**Pull request by rhettg**
See http://docs.aws.amazon.com/kinesis/latest/dev/kinesis-record-processor-additional-considerations.html under Read Throttling.

/cc @bickfordb

### CODE REVIEW

File: triton/stream.go [hunk 1]
```diff
@@ -66,6 +67,32 @@ func (s *ShardStreamReader) wait(minInterval time.Duration) {
...
 }
 
+func isRetryError(err error) bool {
+	if awsErr, ok := err.(awserr.Error); ok {
+		retry := false
+		// Documented Kinesis specific errors as well as common errors we
+		// should probably just retry on.
+		// http://docs.aws.amazon.com/kinesis/latest/APIReference/CommonErrors.html
+		switch awsErr.Code() {
+		case "ProvisionedThroughputExceededException":
```
Comment (bickfordb) [inline triton/stream.go]: `const` these?

**Comment by bickfordb** [comment]
Maybe this should have exponential backoff?

**Comment by bickfordb** [comment]
r+,  seems like this should have backoff or fail eventually?

**Comment by rhettg** [comment]
Yeah, updated with backoff and retry counting.

**Comment by bickfordb** [comment]
looks ok to me if it works

**Pull request merged**